packages/@aws-cdk/aws-apigateway/lib/stage.ts

```diff
@@ -267,3 +280,23 @@ export class Stage extends Resource {
     }
   }
 }
+
+export interface AccessLogSetting {
+  /**
+   * The ARN of the CloudWatch Logs log group or Kinesis Data Firehose delivery stream.
+   * If you specify a Kinesis Data Firehose delivery stream. the stream name must begin
+   * with `amazon-apigateway-`.
+   *
+   * @default - No destination arn
+   */
+  readonly destinationArn?: string;
```

Comment:
Constructs here should not accept an ARN as a string. Instead, this needs to follow the pattern similar to event-sources or destinations.

This means having an interface, something like -

```ts
interface AccessLogDestination {
  bind(): AccessLogDestinationConfig
}

interface AccessLogDestinationConfig {
  readonly destinationArn: string;
}

class CloudwatchLogsDestination implements AccessLogDestination {
   ....
}
```

Take a look at `lambda-event-sources` and `lambda-destinations` modules for examples on how this is done.

Reply:
I understand that you should not receive Arn.

The IAM Role that API Gateway outputs to Logs is common within the account.
Need to be the same as `event-sources` or `destinations`?
Is it not good just to receive logGroup and get Arn?

Reply:
> Is it not good just to receive logGroup and get Arn?

This can be either a log group or a kinesis data firehose delivery stream, and there could be more in the future. Do you see another way to model this?

Reply:
Well, there seems to be no good way.
I'll try it as I told you first.
Thank you.

Reply:
> there seems to be no good way.

I'm not sure I understand what the problem is, from your questions.
Can you detail what you're looking to do, and feel there's no good way to do this?